x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/SpanFlyout/index.tsx

```diff
@@ -74,7 +74,7 @@ function getSpanTypes(span: Span) {
 const SpanBadge = styled(EuiBadge)`
   display: inline-block;
   margin-right: ${px(units.quarter)};
-`;
+` as any;
```

Comment:
This seems a little heavy-handed. Is this the only way to make this work or is there a way we can annotate it to preserve more type information?

I'm ok with merging this as-is but would be interested if knowing we can improve it later.

Reply:
styled-components doesn't interact well with some `a`-vs-`button` things we do in EUI

I can update to `as any as ComponentType<EuiBadgeProps>`, which will preserve the component's props. I did try without the intermediary any but that yields this fun error

> Type instantiation is excessively deep and possibly infinite.

If you don't mind, I'll come back in a follow up PR to this and a few other locations I've `any`ed a styled component.